department-of-veterans-affairs/vets-website
# Platform array builder item page submit on edit

Are you removing, renaming or moving a folder in this PR?

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary
When editing an item in the array builder pattern, use the `submit="prevent"` attribute of the `va-button` to trigger form validation when continuing.

## Issue
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/95567

## Related issue(s)
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2379

## How to run in local environment
1. Check out this branch locally
2. Run `vets-website` and `vets-api`
3. Go to `http://localhost:3001/pension/apply-for-veteran-pension-form-21p-527ez/` in your browser

## How to verify
1. Go to `/mock-form-patterns/employers`
2. Add a new employer
3. 'Edit' the employer and remove a required field(s) on one of the item pages
4. 'Continue' and verify that form validation is triggered and the required field(s) are highlighted and form data is not saved and the next page is not shown
5. Enter required fields(s) and verify that form data is saved and the next page is shown

## Screenshots
### Desktop
#### Edit an array item
![localhost_3001_mock-form-patterns_employers_0_name-and-address_edit=true (1)](https://github.com/user-attachments/assets/82a62fb8-8124-4646-848a-2202bca7a674)

## What areas of the site does it impact?
[Array Builder Pattern(Multiple responses list & loop)](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/patterns/array-builder/README.md)

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
